### PR TITLE
feat(console): create email service connector

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

- Implement email service connector creation feature
- Move `CreateConnectorForm` to the `components` folder since it's shared with the `Connectors` page and the `ConnectorDetails` page.
- Add a `use-connector-api` hook to share the `createConnector` implementation.

### Todo
- move connector-related API calls into the `use-connector-api` hook

## Testing

image

image

## Checklist

- [ ] `.changeset`
- [ ] unit tests
- [ ] integration tests
- [ ] docs

OR

- [x] This PR is not applicable for the checklist

### CODE REVIEW

File: packages/console/src/pages/Connectors/CreateForm/index.tsx [hunk 1]
```diff
@@ -140,7 +156,15 @@ function CreateForm({ onClose, isOpen: isFormOpen, type }: Props) {
             title="general.next"
             type="primary"
             disabled={!activeFactoryId}
-            onClick={() => {
+            onClick={async () => {
+              if (activeFactoryId === ServiceConnector.Email) {
+                const created = await createEmailServiceConnector();
```
Comment (wangsijie) [inline packages/console/src/pages/Connectors/CreateForm/index.tsx:161]: Could you explain the reason that this is special?

**Comment by charIeszhao** [comment]
Should it be `Built-In` instead of `Built-in`?

File: packages/console/src/pages/Connectors/CreateForm/index.tsx (hunk 1 above)
Comment (xiaoyijun) [reply packages/console/src/pages/Connectors/CreateForm/index.tsx]: updated.

**Review by darcyYe — APPROVED** [review]

**Review by wangsijie — APPROVED** [review]

**Comment by xiaoyijun** [comment]
> Should it be `Built-In` instead of `Built-in`?

In titles, apart from the initial letter and proper nouns, the first letters of other words should be lowercase.

**Pull request merged**